entitlement/src/main/java/org/killbill/billing/entitlement/dao/ProxyBlockingStateDao.java

```diff
@@ -243,7 +243,8 @@ public BlockingState getBlockingStateForService(final UUID blockableId, final Bl
     @Override
     public List<BlockingState> getBlockingActiveForAccount(final VersionedCatalog catalog, @Nullable final LocalDate cutoffDt, final InternalTenantContext context) {
         final List<BlockingState> statesOnDisk = delegate.getBlockingActiveForAccount(catalog, cutoffDt, context);
-        return addBlockingStatesNotOnDisk(statesOnDisk, catalog, cutoffDt, context);
+        final List<BlockingState> result = (statesOnDisk.size() > 0) ? addBlockingStatesNotOnDisk(statesOnDisk, catalog, cutoffDt, context) : statesOnDisk;
```

Comment:
I suspect this is the _wip_ part, but it looks like we can do better in `addBlockingStatesNotOnDisk` (e.g. add `IN` clause to avoid pulling unneeded subscriptions).